### PR TITLE
Update target frameworks

### PR DESCRIPTION
**Pull request by jaredpar**
A couple of projects that are included in source build weren't using `$(NetRoslynSourceBuild)`.

### CODE REVIEW

**Review by jasonmalinowski — APPROVED** [review]

**Comment by mthalman** [comment]
The difference between this and a working build of these projects is that [`IsCrossTargetingBuild` gets set to `true` with these changes](https://github.com/dotnet/sdk/blob/9052fef5d4202417e42f5bebdf0bcad94d29fb0a/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.targets#L15-L17). That causes a whole lot less set of targets files from being imported, including the one that provides the `GetTargetPath` target.

**Comment by jaredpar** [comment]
We have many projects that use this TF property, non have this build error. What are we missing?

Also ... could this error be any harder to decipher?

**Comment by mthalman** [comment]
I think the difference with these projects compared to the others is that these are `Exe` output type, not `Library`. I've confirmed that those other projects do not have the `GetTargetPath` target either which means that nothing ever attempts to call that target for those projects.

**Comment by mthalman** [comment]
Does it make sense for an Exe project to be multi-targeting?

**Comment by jaredpar** [comment]
> Does it make sense for an Exe project to be multi-targeting?

How else can we ship csc if we don't multi-target?

**Comment by jasonmalinowski** [comment]
Just to write it here: the story is more fun. There's an implicit ProjectReference being created because there is a project-to-project dependency set in the solution file; that implicit ProjectReference is triggering the invocation to GetTargetPath here which is unexpected, since it's not passing a TFM.

**Comment by jaredpar** [comment]
> Prior to this change it wasn't multi-targeting.

csc's been multi-targeting for years.

**Comment by jaredpar** [comment]
> > Prior to this change it wasn't multi-targeting.

> csc's been multi-targeting for years.

Expanding on this. Our code base must support a variety of runtimes. At the moment we support `net472`, `net6.0`, `net7.0` and `net8.0` depending on what layer we are in. Many of our exes have to support several different runtimes. As such multi-targeting exes is a fact of life for us.

File: src/CodeStyle/CSharp/CodeFixes/Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.csproj [hunk 1]
```diff
@@ -18,6 +18,10 @@
     <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);_GetFilesToPackage</TargetsForTfmSpecificContentInPackage>
     <!-- Remove once https://github.com/NuGet/Home/issues/8583 is fixed -->
     <NoWarn>$(NoWarn);NU5128</NoWarn>
+
+    <!-- Don't inject a project reference for solution build order dependency on CodeStyleConfigFileGenerator
+         because (when in BuildProjectReferences=false) that would call GetTargetPath on its OUTER build which doesn't have such a target -->
+    <AddSyntheticProjectReferencesForSolutionDependencies>false</AddSyntheticProjectReferencesForSolutionDependencies>
```
Comment (jasonmalinowski) [inline src/CodeStyle/CSharp/CodeFixes/Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.csproj:24]: @rainersigwald Which bug do you want to use to track this workaround? This way we could eventually remove the hack.

**Pull request merged**